rep-2004.rst

```diff
@@ -0,0 +1,527 @@
+
+Quality Level 1
+^^^^^^^^^^^^^^^
+
+This category should be used for packages which are required for a reasonable ROS system in a production environment.
+That is to say that after you remove development tools, build tools, and introspection tools, these packages are still left over as requirements for a basic ROS system to run.
+However, that does not mean that packages that would not normally fit this description should never be called 'Level 1'.
+If there is a need for a particular package in a reasonable production scenario, then that package should be considered for this category as well.
+However, packages which we consider essential to getting a robot up and running quickly, but perhaps is a generic solution to the problem should probably not start out as 'Level 1' due to the high effort in getting a package to 'Level 1' and maintaining it there.
+
+For example, the packages which provide intra-process communication, inter-process communication, generated message runtime code, node lifecycle, etc. should probably all be considered for 'Level 1'.
+However, a package which provides pose estimation (like ``robot_pose_ekf``\ ) is a generic solution for something that most people need, but is often replaced with a domain specific solution in production, and therefore it should probably not start out as 'Level 1'.
+However, it may upgrade to it at a later date, if it proves to be a solution that people want to use in their products.
+
+Tools, like ``rostopic``\ , generally do not fall into this category either, but are not categorically excluded.
+For example, it may be the case the tool which launches and verifies a ROS graph (``ros2launch``\ ) may need to be considered 'Level 1' for use in production systems.
+
+Package Requirements
+~~~~~~~~~~~~~~~~~~~~
+
+*Note: bullets below that start with [ROS Core], will be the prescription for what we do in the core packages in order to meet the associated requirements*
+
+Requirements to be considered a 'Level 1' package:
+
+* Version Policy:
+
+  * Must have a version policy (e.g. ``semver``)
+  * Must be at a stable version (e.g. for ``semver`` that means have a version >= 1.0.0)
+  * Must have a strictly declared public API
+  * Must have a policy for API stability
+  * Must have a policy for ABI stability
+  * Must have a policy that keeps API and ABI stability within a released ROS Distribution
+  * [ROS Core] will use ``semver``, will maintain API and ABI stability according to ``semver`` and will be ABI (and therefore API) stable within a ROS distribution
+
+* Change Control Process:
+
+  * Must have all code changes occur through a change request (e.g. pull request, merge request, etc.)
+  * Must have peer review policy for all change requests (e.g. require one or more reviewer)
+  * Must have Continuous Integration (CI) policy for all change requests
+  * Must have documentation policy for all change requests
```

Comment:
I think it should not require anything specific. Projects may wish to do documentation updates each release instead of each change, accumulating the needed changes in a list instead or something.

For the ROS Core packages, I would say we should require them be made at each change (unless there's a reason to wait for an imminent next pull request).